pkg/registry/search/storage/cache.go

```diff
@@ -71,13 +71,21 @@ func (r *SearchREST) newCacheHandler(info *genericrequest.RequestInfo, responder
 		// TODO: process opts.Limit to prevent the client from being unable to process the response
 		// due to the large size of the response body.
 		objItems := r.getObjectItemsFromClusters(clusters, resourceGVR, info.Namespace, info.Name, label)
+		kind, err := r.restMapper.KindFor(resourceGVR)
+		if err != nil {
+			rw.WriteHeader(http.StatusBadRequest)
```

Comment:
Should a StatusInternalServerError be returned here?

Reply:
Yes, it's more appropriate to use 500.